unitary/examples/quantum_checkers/checkers.py

```diff
@@ -0,0 +1,311 @@
+        elif len(self.board.calculate_all_possible_moves(Colors.BLACK)) == 0:
+            return CheckersResult.BLACK_WINS
+        else:
+            return CheckersResult.UNFINISHED
+
+    def do_move(self, move: Move):
+        self.board.move_piece(
+            move.start_row, move.start_col, move.end_row, move.end_col
+        )
+        print(move.start_row, move.start_col, move.end_row, move.end_col)
+
+
+class Square:
+    def __init__(self) -> None:
+        self.occupant = None
+
+
+class Board:
+    def __init__(self, num_rows, num_cols, num_rows_pieces) -> None:
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+
+        # Initalize empty board
+        # test = Square()
+        self.board_matrix = [
+            [Square() for x in range(self.num_cols)] for x in range(self.num_rows)
+        ]
+        if num_rows_pieces * 2 >= num_rows:
+            print(
+                f"Too many rows ({num_rows_pieces}) filled with pieces. Decrease this number for this size of board. [{num_rows}]x[{num_cols}]"
+            )
+            exit()
+
+        # Initialize pieces on correct squares
+        for y in range(num_rows_pieces):
+            for x in range(self.num_cols):
+                if y % 2 == 0 and x % 2 == 0:
+                    self.board_matrix[y][x].occupant = Piece(Colors.BLACK)
+                    self.board_matrix[self.num_rows - 1 - y][x].occupant = Piece(
+                        Colors.WHITE
+                    )
+
+                elif y % 2 != 0 and x % 2 != 0:
+                    self.board_matrix[y][x].occupant = Piece(Colors.BLACK)
+                    self.board_matrix[self.num_rows - 1 - y][x].occupant = Piece(
+                        Colors.WHITE
+                    )
+
+        # self.board_matrix[1][3].occupant = Piece(Colors.WHITE)
+        # Test to see if king works
+        # self.board_matrix[4][4].occupant = Piece(Colors.BLACK, king=True)
+        # self.board_matrix[4][5].occupant = Piece(Colors.WHITE)
+        # self.board_matrix[3][3].occupant = Piece(Colors.WHITE)
+        # self.board_matrix[3][5].occupant = Piece(Colors.WHITE)
+
+    def move_piece(self, from_row, from_col, to_row, to_col):
+        """
+        Moves a piece from given row and column to given row and column. Changes it into a king if it reaches the end
+        """
+        # self.board_matrix[to_row][to_col].occupant = deepcopy(self.board_matrix[from_row][from_col].occupant)
+        self.board_matrix[to_row][to_col].occupant = Piece(
+            self.board_matrix[from_row][from_col].occupant.color,
+            self.board_matrix[from_row][from_col].occupant.king,
+        )
+        self.remove_piece(from_row, from_col)
+
+        # If jumping over a piece, we need to remove it aswell
+        # Jump from column 2 over 3 to 4 we add 3+(3-2)
+        # Jump from column 3 over 2 to 1 we add 2+(2-3)
+        # jump_row = i.end_row+(i.end_row-i.start_row)
+        if abs(to_row - from_row) > 1:
```

Comment:
I assume this does not handle "double jumps"?  Should we verify that abs(to_row - from_row) == 2 and abs(to_col - from_col)==2?